aten/src/ATen/native/mkldnn/Matmul.cpp

```diff
@@ -209,14 +219,28 @@ bool use_mkldnn_bf16_matmul(
     const Tensor& mat1,
     const Tensor& mat2,
     const Tensor& result) {
-  return (
-      use_mkldnn_bf16_matmul() &&
-      mat1.scalar_type() == kBFloat16 &&
-      mat2.scalar_type() == kBFloat16 &&
-      (!result.defined() || result.scalar_type() == kBFloat16) &&
-      mat1.numel() != 0 &&
-      mat2.numel() != 0 &&
-      checksize(mat1, mat2));
+  if (mkldnn_bf16_device_check_arm()) {
```

Comment:
Can aarch64 specific code be guarded under `#if defined(__aarch64__)||defined(__arm64__)`